draft-ietf-quic-transport.md

```diff
@@ -2428,18 +2417,28 @@ A stream in the "open" state may be used by both peers to send frames of any
 type.  In this state, endpoints can send MAX_STREAM_DATA and MUST observe the
 value advertised by its receiving peer (see {{flow-control}}).
 
-From this state, either endpoint can send a frame with the FIN flag set, which
-causes the stream to transition into one of the "half-closed" states.  An
-endpoint sending an FIN flag causes the stream state to become "half-closed
-(local)".  An endpoint receiving a FIN flag causes the stream state to become
-"half-closed (remote)" once all preceding data has arrived.  The receiving
+Opening a stream causes all lower-numbered streams in the same direction to
+become open.  Thus, opening an odd-numbered stream causes all "idle",
+odd-numbered streams with a lower identifier to become open and the same applies
+to even numbered streams.  Endpoints open streams in increasing numeric order,
+but loss or reordering can cause packets that open streams to arrive out of
+order.
+
+From the "open" state, either endpoint can send a frame with the FIN flag set,
+which causes the stream to transition into one of the "half-closed" states.
+This flag can be set on the frame that opens the stream, which causes the stream
+to immediately become "half-closed".  Once an endpoint has completed sending all
+stream data and a STREAM frame with a FIN flag, including any retransmissions,
+the stream state becomes "half-closed (local)".  When an endpoint receives all
+stream data a FIN flag the stream state becomes "half-closed (remote)".  An
 endpoint MUST NOT consider the stream state to have changed until all data has
-arrived.
+either been acknowledged or received.
```

Comment:
I'm not sure if it is obvious that a RST is also considered acknowledged data here.